src/server/repositories/types.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * A type that represents Urls stored in the data store.
+ */
+type StorableUrl = {
+  shortUrl: string
+  longUrl: string
+  state: StorableUrlState
+  clicks: number
+  isFile: boolean
+}
+
```

Comment:
can we take a look at combining this with the existing `UrlType` interface specified in `src/server/models/url.ts`? it looks like we are violating the DRY principle here.

Reply:
UrlType exposes Sequelize methods and making the interface use StorableUrl's fields does not work either since state in `UrlType` is a `Sequelize.EnumDataType<string>`

Reply:
i see. my primary motivation is to avoid the constant typecasting to _unknown_ and then again to _StorableUrl_.

Reply:
Seems like the issue was that the original UrlType had the wrong typings, i.e the runtime type was actually just a number and string but they were labelled differently. Fixing this has allowed converting to the dto type without any typecasting.

Reply:
brilliant